cmd/ncproxy/hcn_networking_test.go

```diff
@@ -0,0 +1,1200 @@
+	}
+	return network.Create()
+}
+
+func TestAddNIC_HCN(t *testing.T) {
+	ctx := context.Background()
+
+	var (
+		containerID      = t.Name() + "-containerID"
+		testNICID        = t.Name() + "-nicID"
+		testEndpointName = t.Name() + "-endpoint"
+	)
+
+	networkingStore, closer, err := createTestNetworkingStore()
+	if err != nil {
+		t.Fatalf("failed to create a test ncproxy networking store with %v", err)
+	}
+	defer closer()
+
+	// setup test ncproxy grpc service
+	agentCache := newComputeAgentCache()
+	gService := newGRPCService(agentCache, networkingStore)
+
+	// test network
+	network, err := createTestNATNetwork(t.Name() + "network")
+	if err != nil {
+		t.Fatalf("failed to create test network with %v", err)
+	}
+	defer func() {
+		_ = network.Delete()
+	}()
+
+	endpoint, err := createTestEndpoint(testEndpointName, network.Id)
+	if err != nil {
+		t.Fatalf("failed to create test endpoint with %v", err)
+	}
+	// defer cleanup in case of error. ignore error from the delete call here
+	// since we may have already successfully deleted the endpoint.
+	defer func() {
+		_ = endpoint.Delete()
+	}()
+
+	// create mocked compute agent service
+	computeAgentCtrl := gomock.NewController(t)
+	defer computeAgentCtrl.Finish()
+	mockedService := computeagentMock.NewMockComputeAgentService(computeAgentCtrl)
+	mockedAgentClient := &computeAgentClient{nil, mockedService}
+
+	// put mocked compute agent in agent cache for test
+	if err := agentCache.put(containerID, mockedAgentClient); err != nil {
+		t.Fatal(err)
+	}
+
+	// setup expected mocked calls
+	mockedService.EXPECT().AddNIC(gomock.Any(), gomock.Any()).Return(&computeagent.AddNICInternalResponse{}, nil).AnyTimes()
+
+	type config struct {
+		name          string
+		containerID   string
+		nicID         string
+		endpointName  string
+		errorExpected bool
+	}
+	tests := []config{
+		{
+			name:          "AddNIC returns no error",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: false,
+		},
+		{
+			name:          "AddNIC returns error with blank container ID",
+			containerID:   "",
+			nicID:         testNICID,
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "AddNIC returns error with blank nic ID",
+			containerID:   containerID,
+			nicID:         "",
+			endpointName:  testEndpointName,
+			errorExpected: true,
+		},
+		{
+			name:          "AddNIC returns error with blank endpoint name",
+			containerID:   containerID,
+			nicID:         testNICID,
+			endpointName:  "",
+			errorExpected: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(_ *testing.T) {
+			req := &ncproxygrpc.AddNICRequest{
+				ContainerID:  test.containerID,
+				NicID:        test.nicID,
+				EndpointName: test.endpointName,
+			}
+
+			_, err := gService.AddNIC(ctx, req)
+			if test.errorExpected && err == nil {
+				t.Fatalf("expected AddNIC to return an error")
```

Comment:
Good point. TBH I don't think it matters much since the end state is the same, but I'm going to change to use the test provided in t.Run instead for goodness.